ruby/ql/test/query-tests/security/cwe-352/railsapp/app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ArticlesController < ApplicationController
+  prepend_before_action :user_authored_article?, only: [:delete_authored_article]
+
+  def delete_authored_article
+    article.destroy
+  end
+
+  def article
+    @article ||= Article.find(params[:article_id])
+  end
+
+  def user_authored_article?
+    @article.author_id = current_user.id
+  end
+end
```

Comment:
Is this file here to demonstrate the danger of having memoised values (`current_user`) and weakened CSRF settings? As far as I can tell the tests don't actually depend on this file, right?

Reply:
Yeah, the file is technically not necessary. It's just for some context into how the non-`with: :exception` calls can be vulnerable.